app/controllers/workflow_controller.rb

```diff
@@ -23,11 +23,23 @@ def show_searchbar?
   end
 
   def button
-    if params[:pressed] == "embedded_configuration_script_payload_tag"
+    case params[:pressed]
+    when 'embedded_configuration_script_payload_map_credentials'
+      javascript_redirect(:action => 'map_credentials', :id => params[:miq_grid_checks])
+    when 'embedded_configuration_script_payload_tag'
       tag(self.class.model)
     end
   end
 
+  def map_credentials
+    assert_privileges('embedded_configuration_script_payload_map_credentials')
+    workflow = find_record_with_rbac(self.class.model, params[:id])
```

Comment:
will this raise an error if the record is not found or just return nil?

Reply:
@Fryguy it raises `ActiveRecord::RecordNotFound: Can't access selected records`

Reply:
Perfect.